unittest/test_diskann.cpp

```diff
@@ -827,3 +839,42 @@ TEST_P(DiskANNTest, generate_cache_list_test) {
     knowhere::DiskANNPrepareConfig::Set(cfg, prep_conf_to_test);
     EXPECT_THROW(diskann->Prepare(cfg), knowhere::KnowhereException);
 }
+
+TEST_P(DiskANNTest, result_distance_test) {
```

Comment:
We can add distance check to all other UT instead of adding a brand new one.

we can make the `dis_truth` as a global variables like `ground_truth`, and calculate it at the beginning time.